src/com/boxboat/jenkins/library/credentials/vault/VaultBaseCredential.groovy

```diff
@@ -0,0 +1,26 @@
+package com.boxboat.jenkins.library.credentials.vault
+
+import com.boxboat.jenkins.library.config.BaseConfig
+import com.boxboat.jenkins.library.config.Config
+import com.boxboat.jenkins.library.credentials.ICredential
+import com.boxboat.jenkins.library.vault.Vault
+
+public abstract class VaultBaseCredential extends BaseConfig<VaultBaseCredential> implements ICredential {
+
+    public String vault
+
+    public String path
+
+    def getSecret(def key){
+        Vault v = Config.global.getVault(vault)
+        def secret = ""
+        v.withCredentials{
+            Config.pipeline.withEnv(["VAULT_ADDR=${v.url}"]){
+                secret = Config.pipeline.sh(returnStdout: true, script: """
+                    printf '{{ (vault "${path}" "${key}") }}' | dockcmd vault get-secrets --input-file -
```

Comment:
Could you have used `com.boxboat.pipeline.common.DockcmdGetSecrets` wrapper here? Or maybe we should move some of that functionality into `com.boxboat.library`?

Reply:
Considered adding a `Dockcmd` class with a `getVaultSecret()`. Figured we could split it out as we added more functionality. Can add that for this PR if you think it would be useful for setting the stage for other secret types

Reply:
From offline conversation -- will be added in a future enhancement if needed